internal/copy/copy_dir.go

```diff
@@ -92,34 +92,52 @@ func CopyDir(dst, src string) error {
 		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
 			target, err := os.Readlink(path)
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to read symlink %q: %w", path, err)
 			}
 
-			return os.Symlink(target, dstPath)
+			if err := os.Symlink(target, dstPath); err != nil {
+				return fmt.Errorf("failed to create symlink %q: %w", dstPath, err)
+			}
+			return nil
 		}
 
-		// If we have a file, copy the contents.
-		srcF, err := os.Open(path)
-		if err != nil {
+		if err := copyFile(dstPath, path, info.Mode()); err != nil {
 			return err
 		}
-		defer srcF.Close()
 
-		dstF, err := os.Create(dstPath)
-		if err != nil {
-			return err
-		}
-		defer dstF.Close()
+		return nil
+	}
 
-		if _, err := io.Copy(dstF, srcF); err != nil {
-			return err
+	return filepath.Walk(src, walkFn)
+}
+
+// copyFile copies the contents and mode of the file from src to dst.
+func copyFile(dst, src string, mode os.FileMode) error {
+	srcF, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("failed to open source file %q: %w", src, err)
+	}
+	defer srcF.Close()
+
+	dstF, err := os.Create(dst)
+	if err != nil {
+		return fmt.Errorf("failed to create destination file %q: %w", dst, err)
+	}
+	defer func() {
+		if cerr := dstF.Close(); cerr != nil {
+			err = fmt.Errorf("failed to close destination file %q: %w", dst, cerr)
```

Comment:
I think this statement doesn't do anything. `err` is not going to be overridden here (even if it would, we should be careful to not wipe the original error). We can just leave it as it was (`defer dstF.Close()`) or we can put some logging in here. In any case let's not try to override the `err`.

Reply:
@ollevche I have proceeded with a simple commit that simplify the defer statement for closing the destination file by removing the error handling logic that attempts to override err. Instead, it logs the error if Close() fails.

It maybe not the best solution but it will allow us to get an error in case it fails. If you could please review it would be great.

Reply:
Thanks @ipolyzos! Let's also change `fmt.Printf` to `log.Printf` because we are using `log` across the rest of the codebase.